src/main/java/com/jpr/maintenance/graphql/GraphQLDataFetchers.java

```diff
@@ -2,57 +2,65 @@
 
 import com.jpr.maintenance.database.model.TaskDetailsEntity;
 import com.jpr.maintenance.database.service.TaskDetailsService;
-import com.jpr.maintenance.validation.errors.InputValidationError;
 import com.jpr.maintenance.validation.model.taskdetails.TaskDetails;
+import graphql.GraphQLError;
 import graphql.execution.DataFetcherResult;
-import graphql.schema.DataFetcher;
-import org.springframework.stereotype.Component;
+import graphql.schema.DataFetchingEnvironment;
+import io.vavr.control.Either;
+import lombok.RequiredArgsConstructor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
 
-import java.util.function.Consumer;
+import static com.jpr.maintenance.graphql.GraphQLUtils.saveEntity;
 
-@Component
+@RequiredArgsConstructor
+@Configuration
 public class GraphQLDataFetchers {
-
     private final TaskDetailsService service;
 
-    public GraphQLDataFetchers(TaskDetailsService service) {
-        this.service = service;
-    }
-
     //TODO refactor to return DataFetcherResult
-    public DataFetcher<TaskDetailsEntity> getTaskDetailsById() {
-        return dataFetchingEnvironment -> {
-            String taskId = dataFetchingEnvironment.getArgument("task_id");
-            return service.findById(Integer.valueOf(taskId)).stream().findFirst().orElse(null);
-        };
+    @Bean
+    public DataFetcherWrapper<TaskDetailsEntity> getTaskDetailsById() {
+        return new DataFetcherWrapper<>(
+                "Query",
+                "taskDetailsById",
+                dataFetchingEnvironment -> {
+                    String taskId = dataFetchingEnvironment.getArgument("task_id");
+                    return service.findById(Integer.valueOf(taskId)).stream().findFirst().orElse(null);
+                }
+        );
     }
 
-    public DataFetcher<DataFetcherResult<TaskDetails>> createTaskDetails() {
-        return dataFetchingEnvironment -> {
-            DataFetcherResult.Builder<TaskDetails> resultBuilder = DataFetcherResult.newResult();
-            TaskDetails.of(dataFetchingEnvironment)
-                .peekLeft(handleErrors())
-                .peek(details -> {
-                    service.save(TaskDetailsEntity.of(details));
-                    resultBuilder.data(details);
-                });
-            return resultBuilder.build();
-        };
+    @Bean
+    public DataFetcherWrapper<DataFetcherResult<TaskDetailsEntity>> createTaskDetails() {
+        return new DataFetcherWrapper<>(
+                "Mutation",
+                "createTaskDetails",
+                dataFetchingEnvironment ->
+                        TaskDetails.of(dataFetchingEnvironment)
+                                .flatMap(t -> saveTaskDetails(t, dataFetchingEnvironment))
+                                .fold(
+                                        l -> DataFetcherResult.<TaskDetailsEntity>newResult().error(l).build(),
```

Comment:
I agree that for "real" variables the names should be named. In case of lambda's I actually prefer the shortest name possible to keep the lambda as readable as possible. But it's a matter of taste, also fine to use more meaningful names if you prefer that.